File: src/ffi/alerts/torrent_removed.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type torrent_removed_alert = crate::ffi::ffi::torrent_removed_alert;
    }
    unsafe extern "C++" {
        include!("cpp/alerts/torrent_removed.h");
        include!("lt-rs/src/ffi/mod.rs.h");
        type InfoHashCpp = crate::ffi::ffi::InfoHashCpp;

        unsafe fn torrent_removed_alert_get_info_hashes(
            a: *mut torrent_removed_alert,
        ) -> InfoHashCpp;
    }
}

File: cpp/alerts/torrent_removed.cpp
#include "./torrent_removed.h"

#include "cpp/lt.h"
#include "lt-rs/src/ffi/mod.rs.h"

namespace ltrs {
    InfoHashCpp torrent_removed_alert_get_info_hashes(lt::torrent_removed_alert* a) {
         return info_hash_t_to_info_hash_cpp(a->info_hashes);
    }
}
